services/blockchain-app-registry/shared/utils/downloadRepository.js

```diff
@@ -188,40 +202,65 @@ const syncWithRemoteRepo = async () => {
 		const { lastSyncedCommitHash, latestCommitHash } = await getCommitInfo();
 
 		if (lastSyncedCommitHash === latestCommitHash) {
-			logger.info('Database is already up-to-date');
+			logger.info('Database is already up-to-date.');
 		} else {
 			const diffInfo = await getDiff(lastSyncedCommitHash, latestCommitHash);
 			// TODO: Add a check to filter out non-blockchain apps related files
-			const filesChanged = diffInfo.data.files.map(file => file.filename);
+			// Sorting is necessary to ensure app.json is downloaded before nativetokens.json file
+			// Otherwise the indexing may fail as token metadata indexing is dependant on app metadata
+			const modifiedFiles = diffInfo.data.files.sort();
```

Comment:
This implementation is incorrect and won't achieve the intended effect (as specified in the comments), since `diffInfo.data.files` will be a list of object(s).
Also, sorting/non-sorting won't make a difference because of `concurrency: modifiedFiles.length` in BluebirdPromise.map. All the files will be processed asynchronously and they kickoff simultaneously.

Reply:
Applied.